src/images/local_images.py

```diff
@@ -0,0 +1,110 @@
+        """
+        All Args Constructor
+        :param images_path: Where the Images Should Be Located
+        """
+        super().__init__(images_path)
+        dir_images = (
+            set()
+            .union(images_path.glob("**/*.jpg"))
+            .union(images_path.glob("**/*.jpeg"))
+        )
+        if len(dir_images) == 0:
+            raise FileNotFoundError(f"No Images Found In Path: {images_path}")
+
+        self.images = dict()
+        for image_path in dir_images:
+            with open(image_path) as image:
+                exif_data = image._getexif()
+                gps_data = exif_data.get(EXIF_GPS_TAG)
+
+                latitude_dms = gps_data[GPS.GPSLatitude]
+                latitude_dir = gps_data[GPS.GPSLatitudeRef]
+                longitude_dms = gps_data[GPS.GPSLongitude]
+                longitude_dir = gps_data[GPS.GPSLongitudeRef]
+
+                location = "{} {}m {}s {} {} {}m {}s {}".format(
+                    latitude_dms[0],
+                    latitude_dms[1],
+                    latitude_dms[2],
+                    latitude_dir,
+                    longitude_dms[0],
+                    longitude_dms[1],
+                    longitude_dms[2],
+                    longitude_dir,
+                )
+
+                self.images[image_path] = location
+
+        self.assigned_images = set()
+
+        log.debug("Images in Directory: {}", self.images)
+
+    @override
+    def get_image_from_coordinates(self, latitude: float, longitude: float) -> dict:
+        """
+        Gets an Image for a Set of Coordinates
+        From A Folder of Local Images Using EXIF Data
+        :param latitude: Latitude of the Point to Get an Image for
+        :param longitude: Longitude of the Point to Get an Image for
+        :return: A Dictionary Containing the Image ID, Path, Latitude, Longitude,
+        Residual Distance From Point, and Error if any
+        """
+        log.debug("Get Image From Coordinates: {}, {}", latitude, longitude)
+        results = {
+            "image_lat": None,
+            "image_lon": None,
+            "residual": None,
+            "image_id": None,
+            "image_path": None,
+            "error": None,
+        }
+
+        filtered_images = filter(
+            lambda img: img not in self.assigned_images, self.images.keys()
+        )
+
+        closest = None
+        closest_distance = np.inf
+        for i, image in enumerate(filtered_images):
+            image_coordinates = Point(self.images[image])
+            coordinates = Point(latitude, longitude)
+            residual = distance(
+                coordinates, image_coordinates, ellipsoid=ELLIPSOIDS["WGS-84"]
+            )
+
+            if residual < closest_distance:
+                closest = image
+                closest_distance = residual
+
+        if closest is None and closest_distance == np.inf:
+            log.debug("No Unassigned Images Available")
+            return results
```

Comment:
If I'm understanding correctly, this will always assign an image (as long as there are more images than points) regardless of the distance between the image and the point. Shouldn't we have a maximum distance here? Otherwise we can get images that are very far away from the points being assigned. Mapillary has this as part of the API call, but we don't have something like that here.

I think the easy implementation here would be to set the initial `closest_distance` to some finite value instead of `np.inf`.

Reply:
i write about this in issue #43

Reply:
Added a `max_distance` CLI option and Image Source Interface param to set a max distance between a point and an image location.